config/prow/cluster/oauth2-proxy/helm/values.yaml

```diff
@@ -17,25 +17,29 @@ affinity:
           operator: In
           values:
           - "true"
-  podAntiAffinity:
-    preferredDuringSchedulingIgnoredDuringExecution:
-    - weight: 100
-      podAffinityTerm:
-        labelSelector:
-          matchExpressions:
-          - key: app.kubernetes.io/name
-            operator: In
-            values:
-            - ingress-nginx
-          - key: app.kubernetes.io/instance
-            operator: In
-            values:
-            - ingress-nginx
-          - key: app.kubernetes.io/component
-            operator: In
-            values:
-            - controller
-        topologyKey: kubernetes.io/hostname
+
+topologySpreadConstraints:
+- maxSkew: 1
+  topologyKey: topology.kubernetes.io/zone
+  whenUnsatisfiable: DoNotSchedule
+  labelSelector:
+    matchExpressions:
+    - key: app.kubernetes.io/name
+      operator: In
+      values:
+      - oauth2-proxy
+    - key: app.kubernetes.io/instance
+      operator: In
+      values:
+      - oauth2-proxy
+    - key: app.kubernetes.io/component
+      operator: In
+      values:
+      - authentication-proxy
```

Comment:
Why did you change `controller` to `authentication-proxy`?

Reply:
and here